packages/grid/_modules_/grid/hooks/root/useGridContainerProps.ts

```diff
@@ -71,6 +71,10 @@ export const useGridContainerProps = (
 
       scrollBarSize.y = hasScrollY ? options.scrollbarSize! : 0;
 
+      // We recalculate the scroll x to consider the size of the y scrollbar.
+      hasScrollX = columnsTotalWidth + scrollBarSize.y > windowSizesRef.current.width;
+      scrollBarSize.x = hasScrollX ? options.scrollbarSize! : 0;
```

Comment:
As an alternative solution. Did we consider to apply the same CSS to the scrollbar detector as the displayed scrollbar? Could this be more reliable and target the root cause better?

Reply:
There were 2 issues
1. we were off on the calculation of the width of the scrollbar
2. the getScrollBarSize function didn't work properly